fairseq/training.py

```diff
@@ -88,69 +148,46 @@ def __init__(
-        :param Path|str fairseq_root: File path to the fairseq git for alternative call of fairseq-hydra-train
+        :param tk.Path fairseq_python_exe: File path to the executable for running python
+        :param tk.Path fairseq_root: File path to the fairseq git for alternative call of fairseq-hydra-train
             (no need to install fairseq here)
         :param bool use_cache_manager: enables caching of data given in the manifest with the i6 cache manager
-        :param [tk.Path]|[str]|tk.Path|str zipped_audio_dir: using a bundle file for caching is very slow for large manifests. For
+        :param [tk.Path]|tk.Path zipped_audio_dir: using a bundle file for caching is very slow for large manifests. For
             speeding up the audio file transfer using the cache manager, a zipped audio directory might be provided.
             The zipped audio directory is then used for caching instead and unzipped on the node for training
         """
 
         # Inputs:
-        self.fairseq_hydra_config = fairseq_hydra_config
+        kwargs = locals()
+        del kwargs["self"]
+
         self.command_line_args = command_line_args or []
-        warning_text = "is specified as input arg and in fairseq_hydra_config. We take the input arg"
-        save_interval_config = fairseq_hydra_config.config_dict.get(
-            "checkpoint", {}
-        ).get("save_interval", None)
-        if save_interval is not None and save_interval_config is not None:
-            logging.warning(
-                "'save_interval' {}: {}".format(warning_text, save_interval)
-            )
-        self.save_interval = save_interval or save_interval_config
-        assert (
-            self.save_interval is not None
-        ), "save_interval has to be set explicitly or via fairseq_hydra_config"
-        max_epoch_config = fairseq_hydra_config.config_dict.get("optimization", {}).get(
-            "max_epoch", None
+        stored_epochs = list(range(save_interval, max_epoch, save_interval)) + [
+            max_epoch
+        ]
+
+        self.keep_epochs = (
+            set(stored_epochs) if keep_epochs is None else set(keep_epochs)
         )
-        if max_epoch is not None and max_epoch_config is not None:
-            logging.warning("'max_epoch' {}: {}".format(warning_text, max_epoch))
-        self.max_epoch = max_epoch or max_epoch_config
-        assert (
-            self.max_epoch is not None
-        ), "max_epoch has to be set explicitly or via fairseq_hydra_config"
-        stored_epochs = list(
-            range(self.save_interval, self.max_epoch, self.save_interval)
-        ) + [self.max_epoch]
-        if keep_epochs is None:
-            self.keep_epochs = set(stored_epochs)
-        else:
-            self.keep_epochs = set(keep_epochs)
         self.fairseq_python_exe = (
             fairseq_python_exe
             if fairseq_python_exe is not None
             else getattr(gs, "FAIRSEQ_PYTHON_EXE", None)
```

Comment:
I would not encourage getting the path from the global settings. Maybe use system Python as a default?